diff --git a/drivers/media/usb/uvc/uvc_driver.c b/drivers/media/usb/uvc/uvc_driver.c
index 88e2fc1..6b89eb5 100644
--- a/drivers/media/usb/uvc/uvc_driver.c
+++ b/drivers/media/usb/uvc/uvc_driver.c
@@ -198,6 +198,16 @@ static struct uvc_format_desc uvc_fmts[] = {
 		.guid		= UVC_GUID_FORMAT_H264,
 		.fcc		= V4L2_PIX_FMT_H264,
 	},
+	{
+		.name		= "Microsoft Luminocity 8bit ('2' 0x0 0x0 0x0)",
+		.guid		= UVC_GUID_FORMAT_MS_L8,
+		.fcc		= V4L2_PIX_FMT_MS_L8,
+	},
+	{
+		.name		= "Microsoft Depth 16bit ('P' 0x0 0x0 0x0)",
+		.guid		= UVC_GUID_FORMAT_MS_D16,
+		.fcc		= V4L2_PIX_FMT_MS_D16,
+	},
 };
 
 /* ------------------------------------------------------------------------
diff --git a/drivers/media/usb/uvc/uvcvideo.h b/drivers/media/usb/uvc/uvcvideo.h
index 94de29e..82d8adc 100644
--- a/drivers/media/usb/uvc/uvcvideo.h
+++ b/drivers/media/usb/uvc/uvcvideo.h
@@ -148,6 +148,15 @@
 #define UVC_GUID_FORMAT_H264 \
 	{ 'H',  '2',  '6',  '4', 0x00, 0x00, 0x10, 0x00, \
 	 0x80, 0x00, 0x00, 0xaa, 0x00, 0x38, 0x9b, 0x71}
+/*
+ * /Microsoft Frame Server GUIDs for Depth images
+ */
+#define UVC_GUID_FORMAT_MS_L8 \
+        { '2',  0x0,  0x0,  0x0, 0x00, 0x00, 0x10, 0x00, \
+         0x80, 0x00, 0x00, 0xaa, 0x00, 0x38, 0x9b, 0x71}
+#define UVC_GUID_FORMAT_MS_D16 \
+        { 'P',  0x0,  0x0,  0x0, 0x00, 0x00, 0x10, 0x00, \
+         0x80, 0x00, 0x00, 0xaa, 0x00, 0x38, 0x9b, 0x71}	 
 
 /* ------------------------------------------------------------------------
  * Driver specific constants.
diff --git a/include/uapi/linux/videodev2.h b/include/uapi/linux/videodev2.h
index 9116a63..5e9765c 100644
--- a/include/uapi/linux/videodev2.h
+++ b/include/uapi/linux/videodev2.h
@@ -609,6 +609,9 @@ struct v4l2_pix_format {
 #define V4L2_PIX_FMT_INRI v4l2_fourcc('I',  'N',  'R',  'I') /* 24 Depth/IR 16:8 */
 #define V4L2_PIX_FMT_INVI v4l2_fourcc('I',  'N',  'V',  'I') /* 8 IR */
 #define V4L2_PIX_FMT_RELI v4l2_fourcc('R',  'E',  'L',  'I') /* 8 IR alternating on off illumination */
+#define V4L2_PIX_FMT_MS_L8 v4l2_fourcc('2', 0x0,  0x0,  0x0) /* Microsoift Frame Server Luminocity 8-bit */
+#define V4L2_PIX_FMT_MS_D16 v4l2_fourcc('P', 0x0,  0x0,  0x0) /* Microsoft Frame Server Depth 16-bit */
+
 
 /*  Vendor-specific formats   */
 #define V4L2_PIX_FMT_CPIA1    v4l2_fourcc('C', 'P', 'I', 'A') /* cpia1 YUV */
